python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -361,6 +367,69 @@ def _calc_stats(input):
 
         self._add_figure(name="time_scale", figure=figure, title="Time Scale")
 
+    def plot_latency(self):
+        if self.output_dir is None:
+            return
+
+        if self.host_time_mapper is None:
+            return
+
+        # Read the pose data to get P1 and GPS timestamps.
+        # The message_index will be used to map back to the host times the message data was recieved.
+        # This will then be compared to the GPS time, assuming the host time was synchronized to GPS time.
+        result = self.reader.read(message_types=[PoseMessage], return_message_index=True,
+                                  source_ids=self.default_source_id, **self.params)
+        pose_data = result[PoseMessage.MESSAGE_TYPE]
+
+        valid_idx = ~np.isnan(pose_data.gps_time)
+
+        if np.sum(valid_idx) == 0:
+            return
+
+        p1_time = pose_data.p1_time[valid_idx]
+        gps_time = pose_data.gps_time[valid_idx]
+        message_index = pose_data.message_index[valid_idx]
+
+        last_gps_time = gps_time[-1]
+
+        # Setup the figure.
+        p1_time_axis_str = 'Relative Time' if self.time_axis == 'relative' else 'P1 Time'
+        figure = make_subplots(rows=1, cols=1, print_grid=False, shared_xaxes=True,
+                               subplot_titles=[f'Pose Message Latency'])
+
+        figure['layout'].update(showlegend=True, modebar_add=['v1hovermode'])
+        figure['layout']['xaxis1'].update(title=f"{p1_time_axis_str} (sec)", showticklabels=True)
+        figure['layout']['yaxis1'].update(title="Latency (s)")
+
+        time = p1_time - float(self.t0)
+
+        # Use the last GPS Time to get the offset between UNIX and GPS time. This includes the epoch difference and the
+        # current leap second.
+        gps_posix_offset = gps2unix(last_gps_time) - last_gps_time
+        gps_posix_times = ((gps_time + gps_posix_offset)*1e9).astype('datetime64[ns]')
+
+        host_posix_times = self.host_time_mapper.get_host_timestamps(message_index)
+
+        # NOTE: The difference between host_posix_times and gps_posix_times is a combination of:
+        # 1. The time the positioning engine took to generate the message
+        # 2. The time the message took to be sent over the transport (e.x. TCP)
+        # 3. The time the delay before the host was able to generate the timestamp (may be high if generated in user
+        #    space code)
```

Comment:
Suggest "may be large/noisy if the data was timestamped in user space instead of in hardware or the kernel"